tests/Valit.Tests/Collection/Primitive_Property_Collection_Validation_Tests.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Shouldly;
+using Xunit;
+
+namespace Valit.Tests.Collection
+{
+    public class Primitive_Property_Collection_Validation_Tests
+    {
+        
+        [Fact]
+        public void EnsureFor_Throws_When_Null_RuleFunc_Is_Given()
+        {
+            var exception = Record.Exception(() => {
+                ValitRules<Model>
+                    .Create()
+                    .EnsureFor(m => m.InvalidValuesCollection, (Func<IValitRule<Model,string>,IValitRule<Model,string>>)null);
+            });
+            
+            exception.ShouldBeOfType(typeof(ValitException));
+        }
+
+        [Fact]
+        public void Validation_With_Complete_Strategy_Fails_When_Invalid_Collection_Is_Given()
+        {
+            var result = ValitRules<Model>
+                .Create()
+                .EnsureFor(m => m.InvalidValuesCollection, _=>_
+                    .Required()
+                        .WithMessage("One")
+                    .Email()
+                        .WithMessage("Two"))
+                .For(_model)
+                .Validate();
+            
+            Assert.False(result.Succeeded);
+            result.ErrorMessages.Where(m => m == "One").Count().ShouldBe(1);
```

Comment:
Can be written as `.Count(m => m == "One")`.
Which is a tiny bit cleaner in my opinion 😄